### ISSUE TITLE
`--js-mode release` is not working

### ISSUE THREAD

**Issue by unarist** [VirtusLab/scala-cli#870]
**Version(s)**

Please provide the version(s) of Scala CLI that is affected by this bug

```
$ scala-cli
0.1.3
```

OS: Windows 11

**Describe the bug**

`--js-mode release` seems to not working, and its output is same as `--js-mode dev`:

**To Reproduce**

test.scala
```
object Entry {
    def main(args: Array[String]) = {}
}
```

commands:

```
$ scala-cli clean test.scala

$ scala-cli package hoge.scala --js --js-mode dev -f
Compiling project (Scala 3.1.1, Scala.js)
Warning: Flag -bootclasspath set repeatedly
Warning: Flag -classpath set repeatedly
Compiled project (Scala 3.1.1, Scala.js)
Wrote R:\scala-cli\test.js, run it with
  node .\test.js

$ ls -la test.js
-rw-r--r-- 1 foo 197121 58886 Apr 11 13:54 test.js

$ scala-cli clean test.scala

$ scala-cli package test.scala --js --js-mode release -f
Compiling project (Scala 3.1.1, Scala.js)
Warning: Flag -bootclasspath set repeatedly
Warning: Flag -classpath set repeatedly
Compiled project (Scala 3.1.1, Scala.js)
Wrote R:\scala-cli\test.js, run it with
  node .\test.js

$ ls -la test.js
-rw-r--r-- 1 foo 197121 58886 Apr 11 13:55 test.js
```

**Expected behaviour**

`--js-mode release` outputs smaller output than `--js-mode dev`.
(more precisely, it would be fullLinkJS output in Scala.js I think)

**Comment by armanbilge** [comment]
@unarist and I discussed this issue on Discord, because it affects my project https://github.com/armanbilge/http4s-cli.

A couple additional notes:
1. Other features such as the `using jsHeader "#!/usr/bin/env node\n"` directive seem to be broken.
2. This is a regression since scala-cli 0.1.1, which I used to publish my project to npm. The version on npm has the correct (expected) output.

**Comment by lwronski** [comment]
@unarist @armanbilge Thanks for reporting. From Scala CLI `0.1.3` uses `scala-js-cli` to linking and we have some regression.

In the next version of Scala CLI it should be fixed in #881
